dtfs-central-api/src/repositories/utilisation-reports-repo/utilisation-report-sql.repo.ts

```diff
@@ -1,10 +1,35 @@
 // TODO FN-1853 - rename this to `utilisation-report.repo.ts` when all repo
 //  methods have been migrated from MongoDB to SQL
 import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
-import { UtilisationReportEntity, ReportPeriod } from '@ukef/dtfs2-common';
+import { UtilisationReportEntity, ReportPeriod, UTILISATION_REPORT_RECONCILIATION_STATUS } from '@ukef/dtfs2-common';
+
+export type GetUtilisationReportDetailsOptions = {
+  reportPeriod?: ReportPeriod;
+  excludeNotUploaded?: boolean;
+};
 
 export const UtilisationReportRepo = SqlDbDataSource.getRepository(UtilisationReportEntity).extend({
   async findOneByBankIdAndReportPeriod(bankId: string, reportPeriod: ReportPeriod): Promise<UtilisationReportEntity | null> {
     return await this.findOneBy({ bankId, reportPeriod });
   },
+
+  async findAllByBankId(bankId: string, options?: GetUtilisationReportDetailsOptions): Promise<UtilisationReportEntity[]> {
+    let query = this.createQueryBuilder('report').where('report.bankId = :bankId', { bankId });
+    if (options?.excludeNotUploaded) {
+      query = query
+        .leftJoinAndSelect('report.azureFileInfo', 'azureFileInfo')
```

Comment:
So I think I need the left join so I can reference the azure file info fields in the next line? 
We can't not check for azure file info at all, because for the completed status, the bank may have uploaded a file or they may have not uploaded a file, so we can't just use status to know if a report was actually uploaded.

Reply:
As discussed offline, I have now changed this to not filter on azure file info and leave that to be filtered by portal-api for the previous reports page specifically.

As a side note, if I had left this as using `createQueryBuilder` I actually would have still had to join on the azure file info table to retrieve it or use an option like the following: `.setFindOptions({loadEagerRelations: true})` since it seems by default it does not load in joint entities even if they are noted as eager, which is perhaps us worth keeping in mind if we do use query builder for future methods!